architecture.md

```diff
@@ -0,0 +1,685 @@
+layers of the container stack. In the case of Docker it is handled by `containerd-shim`.
+In the case of CRI-O it is handled by `conmon`. For clarity, for the remainder
+of this document the term "container process reaper" will be used to refer to
+either reaper. As Kata Containers processes run inside their own  virtual machines,
+the container process reaper cannot monitor, control
+or reap them. `kata-runtime` fixes that issue by creating an [additional shim process
+(`kata-shim`)](https://github.com/kata-containers/shim) between the container process
+reaper and `kata-proxy`. A `kata-shim` instance will both forward signals and `stdin`
+streams to the container process on the guest and pass the container `stdout`
+and `stderr` streams back up the stack to the CRI shim or Docker via the container process
+reaper. `kata-runtime` creates a `kata-shim` daemon for each container and for each
+OCI command received to run within an already running container (example, `docker
+exec`).
+
+The container workload, that is, the actual OCI bundle rootfs, is exported from the
+host to the virtual machine.  In the case where a block-based graph driver is
+configured, virtio-scsi will be used. In all other cases a 9pfs virtio mount point
+will be used. `kata-agent` uses this mount point as the root filesystem for the
+container processes.
+
+## Hypervisor
+
+Kata Containers is designed to support multiple hypervisors.  For the 1.0 release,
+Kata Containers uses just [QEMU](http://www.qemu-project.org/)/[KVM](http://www.linux-kvm.org/page/Main_Page)
+to create virtual machines where containers will run:
+
+![QEMU/KVM](arch-images/qemu.png)
+
+### QEMU/KVM
+
+Depending of the host architecture, Kata Containers support various machine types,
+for example `pc` and `q35` on x86 systems and `virt` on ARM systems. Kata Containers'
+default machine type is `pc`. The default machine type and its [`Machine accelerators`](#Machine-accelerators) can
+be changed by editing the runtime [`configuration`](#Configuration) file.
+
+The follow QEMU features are used in Kata Containers to manage resource constraints, improve
+boot time and reduce memory footprint:
+
+- Machine accelerators.
+- Hot plug devices.
+
+Each feature is documented below.
+
+#### Machine accelerators
+
+Machine accelerators are architecture specific and can used to improve the performance
+and enable specific features of the machine types. The following machine accelerators
+are used in Kata Containers:
+
+- nvdimm: This machine accelerator is x86 specific and only supported by `pc` and
+`q35` machine types. `nvdimm` is used to provide the root filesystem as persistent
+memory device to the Virtual Machine.
+
+Although Kata Containers can run with any recent QEMU release, Kata Containers
+boot time, memory footprint and 9p IO are significantly optimized by using a specific
+QEMU version called [`qemu-lite`](https://github.com/kata-containers/qemu/tree/qemu-lite-2.11.0) and
```

Comment:
Do we want to continue mentioning `qemu-lite` I wonder?

Related: https://github.com/kata-containers/documentation/pulls.

Reply:
I think so.